File: 2. Clean Architecture Objective-C/View Controller.xctemplate/UIViewController/___FILEBASENAME___ViewController.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright (c) ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

#import "___FILEBASENAME___ViewController.h"
#import "___FILEBASENAME___BoundaryModels.h"
#import "___FILEBASENAME___Router.h"

@interface ___FILEBASENAMEASIDENTIFIER___ViewController ()
@end

@implementation ___FILEBASENAMEASIDENTIFIER___ViewController
#pragma mark ___FILEBASENAMEASIDENTIFIER___ViewControllerConfigurable
+ (instancetype)viewControllerConfigured {
    return [___FILEBASENAMEASIDENTIFIER___Configurator viewControllerConfigured];
}

#pragma mark Object lifecycle
- (void)awakeFromNib {
    [super awakeFromNib];
    [___FILEBASENAMEASIDENTIFIER___Configurator configureViewController:self];
}

#pragma mark View lifecycle
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self doSomethingOnAppear];
}

#pragma mark Event handling
- (void)doSomethingOnAppear {
    // NOTE: Ask the Interactor to do some work
    ___FILEBASENAMEASIDENTIFIER___Request *request = [[___FILEBASENAMEASIDENTIFIER___Request alloc] init];
    [self.output doSomethingWithRequest:request];
}

#pragma mark Display logic
- (void)displaySomethingWithViewModel:(___FILEBASENAMEASIDENTIFIER___ResponseViewModel *)viewModel {
    // NOTE: Display the result from the Presenter
    
    // nameTextField.text = viewModel.name
}

#pragma mark Pass segue event to router for scene communication.
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    [self.router passDataBetweenScenesOfSegue:segue];
}

@end

File: 1. Clean Architecture Swift/View Controller.xctemplate/UIViewController + View/___FILEBASENAME___ViewController.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright (c) ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

import UIKit

/**
    Class to control the view of the scene.
 */
class ___VARIABLE_sceneName:identifier___ViewController: UIViewController, ___VARIABLE_sceneName:identifier___PresenterOutput {
    var output: ___VARIABLE_sceneName:identifier___InteractorInput!
    var router: ___VARIABLE_sceneName:identifier___Router!
    
    // MARK: Object lifecycle
    init() {
        super.init(nibName: nil, bundle: nil)
        ___VARIABLE_sceneName:identifier___Configurator.inject(dependenciesFor: self)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        ___VARIABLE_sceneName:identifier___Configurator.inject(dependenciesFor: self)
    }
    
    // MARK: View lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        doSomethingOnLoad()
    }
    
    // MARK: Event handling
    func doSomethingOnLoad() {
        // NOTE: Ask the Interactor to do some work
        let request = ___VARIABLE_sceneName:identifier___.Request()
        output.doSomething(request)
    }
    
    // MARK: Display logic
    func displaySomething(_ viewModel: ___VARIABLE_sceneName:identifier___.Response.ViewModel) {
        // NOTE: Display the result from the Presenter
        // nameTextField.text = viewModel.name
    }
}

// MARK: Custom View
extension ___VARIABLE_sceneName:identifier___ViewController {
    var cView: ___VARIABLE_sceneName:identifier___View {
        get {
            return view as! ___VARIABLE_sceneName:identifier___View
        }
        set {
            view = newValue
        }
    }
    var cViewIfLoaded: ___VARIABLE_sceneName:identifier___View? {
        if let viewIfLoaded = viewIfLoaded {
            return .some(viewIfLoaded as! ___VARIABLE_sceneName:identifier___View)
        }
        else {
            return nil
        }
    }
}
